vm-aws-yaml/Pulumi.yaml

```
@@ -0,0 +1,124 @@
+name: vm-aws-yaml
+description: A YAML program to create a virtual machine on AWS
+runtime: yaml
+
+template: 
+    description: A YAML program to create a virtual machine on AWS
+    config:
+        aws:region:
+            description: The AWS region to deploy into
+            default: us-west-2
+        instanceType:
+            description: Instance type of EC2
+            default: t3.micro
+        vpcNetworkCidr:
+            description: Network CIDR to use for new VPC
+            default: 10.0.0.0/16
+
+configuration:
+    instanceType:
+        type: String
+        default: t3.micro
+    vpcNetworkCidr:
+        type: String
+        default: 10.0.0.0/16
+
+resources:  
+    # Create VPC
+    vpc:
+        type: aws:ec2:Vpc
+        properties:
+            cidrBlock: 10.0.0.0/16
+            enableDnsHostnames: true
+            enableDnsSupport: true 
+    # Create an internet gateway
+    gateway:
+        type: aws:ec2:InternetGateway
+        properties:
+            vpcId: ${vpc.id}
+            tags:
+                Name: vm-aws-yaml-ig
+    # Create a subnet that automatically assigns new instances a public IP address.
+    subnet:
+        type: aws:ec2:Subnet
+        properties:
+            vpcId: ${vpc.id}
+            cidrBlock: 10.0.1.0/24
+            mapPublicIpOnLaunch: true
+            tags:
+                Name: vm-aws-yaml-subnet
+    # Create a route table
+    routeTable:
+        type: aws:ec2:RouteTable
+        properties:
+            vpcId: ${vpc.id}
+            routes:
+                - cidrBlock: 0.0.0.0/0
+                  gatewayId: ${gateway.id}
+            tags:
+                Name: vm-aws-yaml-routetable
+    # Associate the route table with the public subnet.
+    routeTableAssociation:
+        type: aws:ec2:RouteTableAssociation
+        properties:
+            subnetId: ${subnet.id}
+            routeTableId: ${routeTable.id}
+    # Create EC2 Security Group
```

I think it's useful to be a little more verbose on this one, just for clarity:

```suggestion
    # Create a security group allowing inbound access over port 80 and outbound
    # access to anywhere.
```